### DISCUSSION TITLE
mod on fabric

### DISCUSSION THREAD

**Discussion by vdartsabvile** [FabricMC/fabric-api#3138] [Mod Dev Support]
### Description

I'm trying to add an item, but an error pops up.
Error: C:\Users\Vanechka\Desktop\vdatmod-template-1.20\src\main\java\com\vdcompany\item\vdMODitems.java:17: error: incompatible types: ItemGroup cannot be converted to RegistryKey<ItemGroup>
        ItemGroupEvents.modifyEntriesEvent(itemGroup).register(entries -> entries.add(returnitem));
                                           ^

Code: package com.vdcompany.item;

import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.registry.Registries;
import net.minecraft.registry.Registry;
import net.minecraft.util.Identifier;
import com.vdcompany.vdatmod;
public class vdMODitems {
    public static final Item NEW_ITEM = registreItem("new_item", new Item(new FabricItemSettings()));

    private static Item registreItem(String id, Item item) { return Registry.register(Registries.ITEM, new Identifier(vdatmod.MODID, id), item); }
    private static Item registreItem(String id, Item item, ItemGroup itemGroup) {
        Item returnitem = Registry.register(Registries.ITEM, new Identifier(vdatmod.MODID, id), item);
        ItemGroupEvents.modifyEntriesEvent(itemGroup).register(entries -> entries.add(returnitem));

        return returnitem;

    }

    public static void register () { vdatmod.LOGGER.debug("Registering items for: " + vdatmod.MODID); }
}

**Comment by haykam821** [comment]
As of Minecraft 1.20, item groups use registry keys, meaning the event in Fabric API [was changed accordingly](https://github.com/FabricMC/fabric/commit/eff26386bead5475e3f1d8fb04aa0d114f31e588#diff-104ac83edcc69a0d18ea6da559b6974ae38a340740f679118ba226748b64fa5f). Registry keys for vanilla item groups are available in the `ItemGroups` class.
